src/server/pps/server/api_server.go

```diff
@@ -2550,28 +2540,22 @@ func (a *apiServer) DeletePipeline(ctx context.Context, request *pps.DeletePipel
 }
 
 func (a *apiServer) deletePipeline(ctx context.Context, request *pps.DeletePipelineRequest) (response *types.Empty, retErr error) {
-	// Check if there's an StoredPipelineInfo for this pipeline. If not, we can't
+	// Check if there's a PipelineInfo for this pipeline. If not, we can't
 	// authorize, and must return something here
-	pipelinePtr := pps.StoredPipelineInfo{}
-	if err := a.pipelines.ReadOnly(ctx).Get(request.Pipeline.Name, &pipelinePtr); err != nil {
+	pipelineInfo := &pps.PipelineInfo{}
+	if err := a.pipelines.ReadOnly(ctx).Get(request.Pipeline.Name, pipelineInfo); err != nil {
 		if col.IsErrNotFound(err) {
 			return &types.Empty{}, nil
 		}
 		return nil, err
 	}
 
-	// Get current pipeline info from:
-	// - etcdPipelineInfo
-	// - spec commit in etcdPipelineInfo (which may not be the HEAD of the
-	//   pipeline's spec branch)
-	// - kubernetes services (for service pipelines, etc)
-	pipelineInfo, err := a.inspectPipeline(ctx, request.Pipeline.Name)
-	if err != nil {
-		logrus.Errorf("error inspecting pipeline: %v", err)
-		pipelineInfo = &pps.PipelineInfo{Pipeline: request.Pipeline, OutputBranch: "master"}
-	}
-
+	// Attempt to load pipeline details
 	pachClient := a.env.GetPachClient(ctx)
+	if err := ppsutil.GetPipelineDetails(pachClient, pipelineInfo); err != nil {
```

Comment:
This seems sketchy. If we need the pipeline details to delete, then we should just error. If we don't and just default to master, then we would have an incorrect pipeline info when the output branch isn't master.